tfhe/src/high_level_api/integers/unsigned/compressed.rs

```diff
@@ -49,26 +50,57 @@ where
 {
     pub(in crate::high_level_api::integers) ciphertext: CompressedRadixCiphertext,
     pub(in crate::high_level_api::integers) id: Id,
+    pub(crate) tag: Tag,
+}
+
+impl<Id> Tagged for CompressedFheUint<Id>
+where
+    Id: FheUintId,
+{
+    fn tag(&self) -> &Tag {
+        &self.tag
+    }
+
+    fn tag_mut(&mut self) -> &mut Tag {
+        &mut self.tag
+    }
 }
 
 impl<Id> CompressedFheUint<Id>
 where
     Id: FheUintId,
 {
-    pub(in crate::high_level_api::integers) fn new(inner: CompressedRadixCiphertext) -> Self {
+    pub(in crate::high_level_api) fn new(inner: CompressedRadixCiphertext, tag: Tag) -> Self {
         Self {
             ciphertext: inner,
             id: Id::default(),
+            tag,
+        }
+    }
+
+    pub fn into_raw_parts(self) -> (CompressedRadixCiphertext, Id, Tag) {
+        let Self {
+            ciphertext,
+            id,
+            tag,
+        } = self;
+        (ciphertext, id, tag)
+    }
+
+    pub fn from_raw_parts(ciphertext: CompressedRadixCiphertext, id: Id, tag: Tag) -> Self {
+        Self {
+            ciphertext,
+            id,
+            tag,
         }
     }
 
-    pub fn into_raw_parts(self) -> (CompressedRadixCiphertext, Id) {
-        let Self { ciphertext, id } = self;
-        (ciphertext, id)
+    pub fn tag(&self) -> &Tag {
+        &self.tag
     }
 
-    pub fn from_raw_parts(ciphertext: CompressedRadixCiphertext, id: Id) -> Self {
-        Self { ciphertext, id }
+    pub fn tag_mut(&mut self) -> &mut Tag {
+        &mut self.tag
     }
 }
```

Comment:
not in the Tagged trait impl ? probably worth a check for all the fn tag(&self) functions